eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,404 @@
+
+    def setUp(self):
+        """
+        Set up the test suite.
+        """
+        self.default_site = self.get_tenant_data()
+        self.tenant_x = self.get_tenant_data("tenant-x")
+
+    def get_tenant_data(self, prefix: str = "") -> dict:
+        """
+        Get the tenant data.
+
+        If no prefix is provided, the default site data is returned.
+
+        Args:
+            prefix (str): The tenant prefix.
+
+        Returns:
+            dict: The tenant data.
+        """
+        domain = f"{prefix}.local.edly.io" if prefix else "local.edly.io"
+        return {
+            "base_url": f"http://{domain}",
+            "domain": domain,
+        }
+
+    def get_access_token(self, tenant_base_url: str) -> str:
+        """
+        Get an access token for a tenant.
+
+        Args:
+            tenant_base_url (str): The tenant base URL.
+
+        Returns:
+            str: The access token.
+        """
+        data = {
+            "client_id": CLIENT_ID,
+            "client_secret": CLIENT_SECRET,
+            "grant_type": "client_credentials",
+        }
+        url = f"{tenant_base_url}/oauth2/access_token/"
+        response = requests.post(url, data=data, timeout=API_TIMEOUT)
+        return response.json()["access_token"]
+
+    # pylint: disable=too-many-arguments
+    def make_tenant_request(
+        self,
+        tenant: dict,
+        method: str,
+        url: str,
+        data: dict | None = None,
+        params: dict | None = None,
+    ) -> HttpResponse:
+        """
+        Make a request to a tenant.
+
+        Args:
+            tenant (dict): The tenant data.
+            method (str): The HTTP method ('GET', 'POST', etc.).
+            url (str): The URL to make the request to.
+            data (dict, optional): The data for POST, PATCH and PUT requests.
+            params (dict, optional): The parameters for GET and DELETE requests.
+
+        Returns:
+            HttpResponse: The response object.
+        """
+        access_token = self.get_access_token(tenant["base_url"])
+        headers = {"Host": tenant["domain"], "Authorization": f"Bearer {access_token}"}
+        full_url = f"{tenant['base_url']}{url}"
+
+        method = method.upper()
+        if method == "POST":
+            response = requests.post(full_url, data=data, headers=headers, timeout=API_TIMEOUT)
+        elif method == "GET":
+            response = requests.get(full_url, params=params, headers=headers, timeout=API_TIMEOUT)
+        elif method == "PATCH":
+            response = requests.patch(full_url, json=data, headers=headers, timeout=API_TIMEOUT)
+        else:
+            raise ValueError(f"Unsupported HTTP method: {method}")
+
+        return response
+
+
+@ddt
+class TestUsersAPIIntegration(BaseAPIIntegrationTest):
+    """Integration test suite for the Users API"""
+
+    def setUp(self):
+        """Set up the test suite"""
+        self.user_url = "/eox-core" + reverse("eox-api:eox-api:edxapp-user")
```

Comment:
@BryanttV, since we will be running the tests outside the LMS container in an isolated environment, we won’t need to use reverse

Reply:
I think one of the purposes of using `reverse` is in case the URL changes at some point. But, we can directly define the string.

Reply:
Mmm, that's a fair point. Let's try and see if it works with reverse then